commands/databases.go

```diff
@@ -1152,3 +1154,308 @@ func RunDatabaseSetSQLModes(c *CmdConfig) error {
+
+	cmdDatabaseFirewallUpdate := CmdBuilder(cmd, RunDatabaseFirewallRulesUpdate, "update <db-id> --rules type:value [--rule type:value]", "Update the firewall rules for a given database. The rules passed in to the --rules flag will replace the firewall rules previously assigned to the database,", databaseFirewallUpdateDetails,
+		Writer, aliasOpt("u"))
+	AddStringSliceFlag(cmdDatabaseFirewallUpdate, doctl.ArgDatabaseFirewallRule, "", []string{}, databaseFirewallRulesTxt, requiredOpt())
+
+	cmdDatabaseFirewallCreate := CmdBuilder(cmd, RunDatabaseFirewallRulesCreate, "add <db-id> --rule type:value", "Add a database firewall rule to a given database", databaseFirewallAddDetails,
+		Writer, aliasOpt("a"))
+	AddStringFlag(cmdDatabaseFirewallCreate, doctl.ArgDatabaseFirewallRule, "", "", "", requiredOpt())
+
+	cmdDatabaseFirewallRemove := CmdBuilder(cmd, RunDatabaseFirewallRulesRemove, "remove <firerule-uuid>", "Remove a firewall rule for a given database", databaseFirewallRemoveDetails,
+		Writer, aliasOpt("r"))
+	AddStringFlag(cmdDatabaseFirewallRemove, doctl.ArgDatabaseFirewallRuleUUID, "", "", "", requiredOpt())
+
+	return cmd
+
+}
+
+// displayDatabaseFirewallRules calls Get Firewall Rules to list all current rules.
+func displayDatabaseFirewallRules(c *CmdConfig, short bool, id string) error {
+	firewallRules, err := c.Databases().GetFirewallRules(id)
+	if err != nil {
+		return err
+	}
+
+	item := &displayers.DatabaseFirewallRules{
+		DatabaseFirewallRules: firewallRules,
+	}
+
+	return c.Display(item)
+}
+
+// All firewall rules require the databaseID
+func firewallRulesArgumentCheck(c *CmdConfig) error {
+	if len(c.Args) == 0 {
+		return doctl.NewMissingArgsErr(c.NS)
+	}
+	if len(c.Args) > 1 {
+		return doctl.NewTooManyArgsErr(c.NS)
+	}
+	return nil
+}
+
+// RunDatabaseFirewallRulesList retrieves a list of firewalls for specific database cluster
+func RunDatabaseFirewallRulesList(c *CmdConfig) error {
+	err := firewallRulesArgumentCheck(c)
+	if err != nil {
+		return err
+	}
+
+	id := c.Args[0]
+
+	return displayDatabaseFirewallRules(c, true, id)
+}
+
+// RunDatabaseFirewallRulesUpdate replaces previous rules with the rules passed in to --rules
+func RunDatabaseFirewallRulesUpdate(c *CmdConfig) error {
+	err := firewallRulesArgumentCheck(c)
+	if err != nil {
+		return err
+	}
+
+	id := c.Args[0]
+	r, err := buildDatabaseUpdateFirewallRulesRequestFromArgs(c)
+	if err != nil {
+		return err
+	}
+
+	err = c.Databases().UpdateFirewallRules(id, r)
+	if err != nil {
+		return err
+	}
+
+	return displayDatabaseFirewallRules(c, true, id)
+
+}
+
+// buildDatabaseUpdateFirewallRulesRequestFromArgs will ingest the --rules arguments into a DatabaseUpdateFirewallRulesRequest object.
+func buildDatabaseUpdateFirewallRulesRequestFromArgs(c *CmdConfig) (*godo.DatabaseUpdateFirewallRulesRequest, error) {
+	r := &godo.DatabaseUpdateFirewallRulesRequest{}
+
+	firewallRules, err := c.Doit.GetStringSlice(c.NS, doctl.ArgDatabaseFirewallRule)
```

Comment:
that's a really good thought. After some tests, I discovered I would have to muck with repos outside of doctl (folders in the vender/github.com). I know ben was trying to avoid doing that as I've already had to do make changes to godo.